.github/workflows/release.yml

```diff
@@ -0,0 +1,47 @@
+on:
+  release:
+    types:
+      - published
+
+name: release
+
+permissions:
+  id-token: write
+
+jobs:
+  pypi:
+    name: upload release to PyPI
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+
+    - uses: actions/setup-python@v4
+      with:
+        python-version: ">= 3.7"
+
+    - name: deps
+      run: python -m pip install -U setuptools build wheel
+
+    - name: build
+      run: python -m build
+
+    - name: publish
+      uses: pypa/gh-action-pypi-publish@master
+      with:
+        user: __token__
+        password: ${{ secrets.PYPI_TOKEN }}
+
+    - name: sign
+      uses: trailofbits/gh-action-sigstore-python@v0.0.2
+      with:
+        inputs: ./dist/*.tar.gz ./dist/*.whl
```

Comment:
I threw in the ToB signing action as well; let me know if you'd like to do that separately or not at all.